File: oopl/lab2/q1.java
import java.util.*;

class q1{
	public static void main(String args[])
	{
		Scanner sc = new Scanner(System.in);
		int n, i, j;
		System.out.println("enter number of elements");
		n = sc.nextInt();
		int a[] = new int[n];
		System.out.println("enter elements");
		for(i = 0; i < n; i++)
			a[i] = sc.nextInt();
		for(i = 0; i < n-1; i++)
		{
			for(j=0; j < n-1-i; j++)
			{
				if(a[j] > a[j+1])
				{
					int t = a[j];
					a[j] = a[j+1];
					a[j+1] = t;
				}
			}
		}
		System.out.println("sorted array is:");
		for(i = 0; i < n; i++)
			System.out.print(a[i] + "\t");
	}
}

File: sem-6-labs/itl/lab4/q1.py
class subs:
	def ssets(self, sset):
		return self.subsetsRecur([], sorted(sset))

	def subsetsRecur(self, current, sset):
		if sset:
			return self.subsetsRecur(current, sset[1:]) + self.subsetsRecur(current + [sset[0]], sset[1:])
		return [current]

n = int(input("enter number of elements : "))
arr = list(map(int, input("\nenter the numbers: ").strip().split()))[:n]
print(subs().ssets(arr))